### PR TITLE
Tobias

### PR DESCRIPTION
**Pull request by tobyy22**

### CODE REVIEW

File: pgwidget/pgwidget_core.py [hunk 1]
```diff
@@ -1049,8 +1114,76 @@ def draw_children(self):
...
+                    target_coordinates=(i+1, j)
+
+            target_cell_index=self.find_cell_index(*target_coordinates)
+            if target_cell_index is not None:
+                self.deselect_all_cells()
+                self.select_cell(target_cell_index)
+            else:
+                print("Move cam, branch")
+                self.move_camera(*target_coordinates)
+
```
Comment (DovaX) [inline pgwidget/pgwidget_core.py:1199]: This function needs to be refactored

File: pgwidget/pgwidget_core.py [hunk 2]
```diff
@@ -839,6 +862,37 @@ def get_row_and_col_of_cell(self,cell):
...
+        self.table_cells = []
+        self.cols = cols
+        self.rows = rows
+
+        self._initialize_cells()
+
+
+        self.table_size = [(self.cell_size[0] + self.margin) * self.cols + int(self.has_scrollbar) * 15, (self.cell_size[1] + self.margin) * (self.rows + 1)]
+
```
Comment (DovaX) [inline pgwidget/pgwidget_core.py:882]: 15 = magical constant?

Comment (DovaX) [reply pgwidget/pgwidget_core.py:882]: Fix everywhere please

File: pgwidget/pgwidget_core.py (hunk 1 above)
Comment (DovaX) [reply pgwidget/pgwidget_core.py:1199]: Inside of every "direction" branch should be only several quite generic functions - like "get_next_row_coordinates(...)", "shift_row(...)" -> Try harder please, move_selected can be implemented on max 20 rows of code - focus on remove nested if and else conditions. It is not easy to read right now.

**Review by DovaX — COMMENTED** [review]
Please fix the suggested changes

File: pgwidget/pgwidget_core.py (hunk 2 above)
Comment (tobyy22) [reply pgwidget/pgwidget_core.py:882]: The same initialization of self.table_size is used in the __init__. Should I refactor that everywhere and make the scrollbar width a variable?

**Pull request merged**